features/netsocket/nsapi_types.h

```diff
@@ -73,13 +73,14 @@ enum nsapi_error {
 
 
 /** Enum of event types
- * 
+ *
  *  Event callbacks are accompanied with an event-dependent parameter passed as an intptr_t.
  *
  *  @enum nsapi_event
  */
  typedef enum nsapi_event {
-    NSAPI_EVENT_CONNECTION_STATUS_CHANGE = 0 /*!< network connection status has changed, the parameter = new status (nsapi_connection_status_t) */
+    NSAPI_EVENT_CONNECTION_STATUS_CHANGE = 0, /*!< network connection status has changed, the parameter = new status (nsapi_connection_status_t) */
+    NSAPI_EVENT_CELLULAR_STATUS_CHANGE   = 1  /*!< cellular modem status has changed, the parameter = new status (cellular_connection_status_t) */
```

Comment:
Kevin also suggested that we could request "a chunk of number space like 0x1000-0x10ff from nsapi_event_t to give you a range of events, so each of them can still have a useful parameter".

Reply:
NSAPI_EVENT_CONNECTION_STATUS_CHANGE is already used in this way so we should this in same way for consistence.